lib/Tumblr/StreamBuilder/Streams/BufferedRankedStream.php

```diff
@@ -129,4 +129,12 @@ public static function from_template(StreamContext $context): self
             $context->get_cache_provider()
         );
     }
+
+    /**
+     * @inheritDoc
+     */
+    protected function can_enumerate(): bool
+    {
+        return $this->inner->can_enumerate() && parent::can_enumerate();
```

Comment:
ever so slightly faster, as you're doing in other changes:

```suggestion
        return parent::can_enumerate() && $this->inner->can_enumerate();
```